tests/Feature/SampleFeatureTest.php

```diff
@@ -31,4 +38,288 @@ public function testWithMock()
+                ]),
+                'contacts.name'
+            ],
+            'incorrect sorted list with nested keys and nullable values (by contact.name)' => [
+                new Collection([
+                    ['id' => 1, 'contact' => ['name' => 'Alex']],
+                    ['id' => 2, 'contact' => ['name' => 'Bill']],
+                    ['id' => 4, 'contact' => ['name' => null]],
+                    ['id' => 5, 'contact' => ['name' => null]],
+                    ['id' => 3, 'contact' => ['name' => 'Chuck']],
+                ]),
+                'contacts.name',
+                AssertionFailedError::class
+            ],
+            'correct sorted list with 2 level nested keys(by company.contact.name)' => [
+                new Collection([
+                    ['id' => 3, 'company' => [
+                        'name' => 'Alex inc.',
+                        'contact' => ['name' => 'Alex'],
+                    ]],
+                    ['id' => 4, 'company' => [
+                        'name' => 'Bill inc.',
+                        'contact' => ['name' => 'Bill'],
+                    ]],
+                    ['id' => 1, 'company' => [
+                        'name' => 'Chuck inc.',
+                        'contact' => ['name' => 'Chuck'],
+                    ]],
+                    ['id' => 2, 'company' => [
+                        'name' => 'Tim inc.',
+                        'contact' => ['name' => 'Tim'],
+                    ]],
+                    ['id' => 5, 'company' => [
+                        'name' => 'Wood inc.',
+                        'contact' => ['name' => 'Wood'],
+                    ]],
+                ]),
+                'companies.contacts.name'
+            ],
+        ];
+    }
+
+    /**
+     * Test sorting by several columns.
+     *
+     * @param Collection $resultData Collection of data
+     * @param string[] $sortingFields Sorting fields
+     * @param \Exception|null $expectedException Expected exception
+     *
+     * @dataProvider multiSortingData
+     */
+    public function testMultiSorting(Collection $resultData, array $sortingFields, $expectedException = null)
+    {
+        if ($expectedException) {
+            $this->expectException($expectedException);
+        }
+
+        $sortingString = collect($sortingFields)->implode(',');
+
+        $serviceMock = Mockery::mock(MyService::class);
+        $serviceMock->shouldReceive('getData')->with($sortingString)->andReturn($resultData);
+        $serviceMock->shouldReceive('getData')->with("-".$sortingString)
+            ->andReturn($resultData->reverse());
+
+        $this->app->bind(MyService::class, function () use ($serviceMock) {
+            return $serviceMock;
+        });
+
+        $count = $resultData->count();
+
+        $this->assertMultiSortingWorks("api/test-order-by", $count, $sortingFields, []);
+    }
+
+    /** Data for testing sorting by several fields */
+    public function multiSortingData()
```

Comment:
let's rename it accordingly to `sortByMultipleFieldsExamples()`

Reply:
done.